src/test_l3_dsHdmiIn.c

```diff
@@ -0,0 +1,1461 @@
+      goto exit;
+   }
+
+   printf("\n*******Enter the y coordinate to select*******\t"
+                     "Acceptable inputs are:\t"
+                     "Min is 0\t"
+                     "Max is based on the current resolution\t"
+             "**********************************************\n");
+   scanf("%d",&y);
+   readAndDiscardRestOfLine(stdin);
+   if(y < 0)
+   {
+      printf("\n invalid y coordinate selected \n");
+      goto exit;
+   }
+
+   printf("\n*******Enter the width to select***************\t"
+                     "Acceptable inputs are:\t"
+                     "Min is 0\t"
+                     "Max is based on the current resolution\t"
+            "**********************************************\n");
+   scanf("%d",&width);
+   readAndDiscardRestOfLine(stdin);
+   if(width < 0)
+   {
+      printf("\n invalid width selected \n");
+      goto exit;
+   }
+
+   printf("\n*******Enter the height to select**************\t"
+                     "Acceptable inputs are:\t"
+                     "Min is 0\t"
+                     "Max is based on the current resolution\t"
+             "**********************************************\n");
+   scanf("%d",&height);
+   readAndDiscardRestOfLine(stdin);
+   if(width < 0)
+   {
+      printf("\n invalid height selected \n");
+      goto exit;
+   }
+
+  ret = dsHdmiInScaleVideo(x, y, width, height);
+  if(ret != dsERR_NONE)
+  {
+     UT_LOG_ERROR("Result: Failed to dsHdmiInScaleVideo:[%s]",ut_control_GetMapString(dsError_mapTable, ret));
+  } else {
+     UT_LOG_INFO("Result : Passed dsHdmiInScaleVideo paramsx : %d, y : %d, heitht : %d , width : %d",
+                  x, y , width, height);
+  }
+
+exit:
+UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+/**
+* @brief This test to select zoom mode.
+*
+* This test function is select zoom mode of selected HdmiInput port on platform.
+*
+* **Test Group ID:** 03@n
+* **Test Case ID:** 006@n
+*
+* **Test Procedure:**
+* Refer to Test specification documentation
+* [ds-hdmi-in_halSpec.md](../../docs/pages/ds-hdmi-in_halSpec.md)
+*/
+void test_l3_HdmiIn_zoom_mode(void)
+{
+    gTestID = 6;
+    UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t ret   = dsERR_NONE;
+    int select;
+    dsVideoZoom_t mode;
+
+    printf("\n===================Select Zoom Mode================\n");
+    printf("\n**************Availabe Zoom Modes are****************\n");
+    for(dsVideoZoom_t i = dsVIDEO_ZOOM_NONE ; i < dsVIDEO_ZOOM_MAX ; i++)
+    {
+           UT_LOG_INFO("%s : %d", ut_control_GetMapString(dsVideoZoom_mapTable, i) , i);
+    }
+    printf("\n******************************************************\n");
+    scanf("%d",&select);
+    readAndDiscardRestOfLine(stdin);
+    if(select < 0 || select > dsVIDEO_ZOOM_MAX)
+    {
+      printf("\n invalid zoom mode selected \n");
+      goto exit;
+    }  
+    
+    mode = (dsVideoZoom_t)select;
+    ret = dsHdmiInSelectZoomMode(mode);
+    if(ret != dsERR_NONE)
```

Comment:
Remove the else case

In all cases, you need to see the results output

You've got some serious debugging issues todo if this was to assert and a failure of L1-L2 to fix in that case.

Reply:
done

Reply:
Please refer https://github.com/rdkcentral/rdk-halif-test-device_settings/pull/116 @ishthiyaqahmed